sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -288,15 +290,22 @@ private WindowDefinition getWindowDefinition(Optional<Window> maybeWindow, Expre
             return symbol;
         });
 
-        WindowFrameDefinition windowFrameDefinition = WindowDefinition.DEFAULT_WINDOW_FRAME;
+        WindowFrameDefinition windowFrameDefinition = WindowDefinition.UNBOUNDED_PRECEDING_CURRENT_ROW;
         if (window.getWindowFrame().isPresent()) {
             WindowFrame windowFrame = window.getWindowFrame().get();
             FrameBound start = windowFrame.getStart();
+            if (start.equals(UNBOUNDED_FOLLOWING)) {
```

Comment:
Do we need the checks both here and in the `WindowAgg` operator?

If this code could handle other definitions already as well I think we could keep with one limitation that is closer to the components that cannot handle other definitions.

Reply:
These are sanity checks for window definitions (preventing invalid frame definitions), whereas the WindowAgg prevents unsupported frames (eg. `ROWS`)

> If this code could handle other definitions already as well I think we could keep with one limitation that is closer to the components that cannot handle other definitions.

I don't understand what you mean here.

Reply:
Ah, I didn't realize that this is a semantic invalid frame definition but assumed that it is not yet implemented. 

Should we add tests for these semantic validations then? Seems like there currently aren't any.